sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/CatalogManager.scala

```diff
@@ -44,8 +44,12 @@ class CatalogManager(
   import CatalogV2Util._
 
   private val catalogs = mutable.HashMap.empty[String, CatalogPlugin]
+  private val validNameFormat = "([\\w_]+)".r
 
   def catalog(name: String): CatalogPlugin = synchronized {
+    if (!validNameFormat.pattern.matcher(name).matches()) {
+      throw QueryCompilationErrors.invalidNameForCatalogError(name)
+    }
```

Comment:
The verification of name may be a bit strict. But as an identifier. I think the basic check is needed like follow:
```scala
name.trim.isEmpty
```